update_branch_protections.py

```diff
@@ -0,0 +1,234 @@
+    """
+    Read and parse a repository data list from ``repo_data_path``
+    Expects a basic csv format (comma-delimited, first row are column headers)
+    Expects the following header names
+    - 'owner': repository owner, organization
+    - 'repo_name': repository name
+    - 'repo_type': one of
+      - {0}
+    And a bool-type (y, yes, true) for each of the protection rule types:
+    - {1}
+
+    Parameters
+    ----------
+    repo_data_path : str
+        path to the repo-data csv
+
+    Returns
+    -------
+    List[ProtectionGroup]
+        A list of ProtectionGroup's, holding settings for protection groups
+    """
+    data_path = Path(repo_data_path)
+    if not data_path.exists:
+        print('repo data file does not exist')
+        return
+    repo_data = []
+    # Deal with extra columns
+    with open(data_path, 'r') as csvfile:
+        csv_reader = csv.DictReader(csvfile)
+        for row in csv_reader:
+            data_row = ProtectionGroup.from_dict(row)
+            repo_data.append(data_row)
+    return repo_data
+
+
+# This could be a decorator, but we only do this once here
+parse_repo_list.__doc__ = parse_repo_list.__doc__.format(
+    '\n      - '.join(list(ProtectionGroup.CHECK_NAME_MAP.keys())),
+    '\n    - '.join(ProtectionGroup.RULE_MAP.keys())
+)
+
+
+def main(
+    owner: str = "pcdshub",
+    repo_name: str = "",
+    repo_type: str = "Python Library",
+    prot_master: bool = False,
+    prot_pages: bool = False,
+    prot_default: bool = False,
+    repo_data_path: str = "",
+    write: bool = False
+):
+    if repo_data_path:
+        # run for each repo in list
+        repo_data = parse_repo_list(repo_data_path)
+        for protection_group in repo_data:
+            protection_group.apply_protections(write=write)
+        return
+
+    if not repo_name:
+        return
+
+    # apply settings for a single repo
+    protection_group = ProtectionGroup(
+        owner=owner, repo_name=repo_name, repo_type=repo_type,
+        master=prot_master, gh_pages=prot_pages, default=prot_default
+    )
+    protection_group.apply_protections(write=write)
+
+    return
+
+
+def _create_argparser() -> argparse.ArgumentParser:
+    """
+    Create an ArgumentParser for update_branch_protections
+
+    Returns
+    -------
+    argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser()
+    # specify an owner, repo, and settings
+    parser.add_argument("owner", type=str, default='pcdshub', nargs='?',
+                        help='Organization or owner of the repository, "pcdshub"'
+                             'by default')
+    parser.add_argument("repo_name", type=str, default='', nargs='?',
+                        help='Name of the repository')
+    parser.add_argument("repo_type", type=str, default='Other', nargs='?',
+                        help='Type of the repository.  Should be one of '
+                             f'{list(ProtectionGroup.CHECK_NAME_MAP.keys())}')
```

Comment:
I've never tried this before, but I remembered argparse being able to take care of "must be one of these" sorts of args: https://docs.python.org/3/library/argparse.html#choices